addons/arsenal/functions/fnc_refresh.sqf

```diff
@@ -15,13 +16,37 @@
  *
  * Public: Yes
 */
-params [["_updateItems", true, [true]]];
+params [["_updateItems", true, [true]], ["_updateVirtualItems", false, [false]]];
+
+TRACE_2("",_updateItems,_updateVirtualItems);
+
+// Don't execute in scheduled environment
+if (canSuspend) exitWith {
+    [{_this call FUNC(refresh)}, _this] call CBA_fnc_directCall;
+};
 
 if (_updateItems) then {
     // Update current item list
     call FUNC(updateCurrentItemsList);
 
     // This takes care of unique inventory items (arsenal doesn't have it whitelisted)
+    if (!_updateVirtualItems) then {
+        call FUNC(updateUniqueItemsList);
+    };
+};
+
+private _virtualItems = GVAR(currentBox) getVariable QGVAR(virtualItems);
+if (isNil "_virtualItems") exitWith {
```

Comment:
this ends up being `nil` in 3den when editing a unit's inventory

the displayText and execNextframe both don't work in 3den so I guess it's ok
but it feels a little dangerous

Reply:
I think it might be possible to not run this function at all when in 3den, so we could add a condition at the very top to exit.